packages/build-tools/src/steps/functions/startAndroidEmulator.ts

```diff
@@ -0,0 +1,147 @@
+import assert from 'assert';
+
+import { PipeMode } from '@expo/logger';
+import { BuildFunction, BuildStepInput, BuildStepInputValueTypeName } from '@expo/steps';
+import spawn from '@expo/turtle-spawn';
+import { v4 as uuidv4 } from 'uuid';
+
+import { retryAsync } from '../../utils/retry';
+
+const defaultSystemImagePackage = `system-images;android-30;default;${
+  process.arch === 'arm64' ? 'arm64-v8a' : 'x86_64'
+}`;
+
+export function createStartAndroidEmulatorBuildFunction(): BuildFunction {
+  return new BuildFunction({
+    namespace: 'eas',
+    id: 'start_android_emulator',
+    name: 'Start Android Emulator',
+    inputProviders: [
+      BuildStepInput.createProvider({
+        id: 'device_name',
+        required: false,
+        defaultValue: 'EasAndroidDevice01',
+        allowedValueTypeName: BuildStepInputValueTypeName.STRING,
+      }),
+      BuildStepInput.createProvider({
+        id: 'system_image_package',
+        required: false,
+        defaultValue: defaultSystemImagePackage,
+        allowedValueTypeName: BuildStepInputValueTypeName.STRING,
+      }),
+    ],
+    fn: async ({ logger }, { inputs }) => {
+      const deviceName = `${inputs.device_name.value}`;
+      const systemImagePackage = `${inputs.system_image_package.value}`;
+      logger.info('Making sure system image is installed');
+      await spawn('sdkmanager', [systemImagePackage], {
+        logger,
+      });
+
+      logger.info('Creating emulator device');
+      const avdManager = spawn(
+        'avdmanager',
+        ['create', 'avd', '--name', deviceName, '--package', systemImagePackage, '--force'],
+        {
+          stdio: 'pipe',
+        }
+      );
+      avdManager.child.stdin?.write('no');
```

Comment:
What does it do?

Reply:
It answers `avdmanager`'s question whether we want to create a custom build profile for the AVD. I'll add a comment.

<img width="659" alt="Zrzut ekranu 2024-01-30 o 12 23 37" src="https://github.com/expo/eas-build/assets/1151041/415b0ae1-e75a-4ea8-baaa-e424638b33d9">

Reply:
Added comment.